scipy/signal/tests/test_bsplines.py

```diff
@@ -0,0 +1,214 @@
+# pylint: disable=missing-docstring
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+from numpy import array, nan
+from numpy.testing import (TestCase, assert_raises, run_module_suite,
+                           assert_allclose, assert_array_equal,
+                           assert_equal, assert_almost_equal)
+
+import scipy.signal.bsplines as bsp
+
+class TestBSplines(TestCase):
+
+    def test_factorial(self):
+        # can't all be zero state
+        assert_equal(bsp.factorial(1), 1)
+
+    def test_spline_filter(self):
+        np.random.seed(12457)
+        # Test the type-error branch
+        assert_raises(TypeError, bsp.spline_filter, array([0]), 0)
+        # Test the complex branch
+        data_array_complex = np.random.rand(7, 7) + np.random.rand(7, 7)*1j
+        # make the magnitude exceed 1, and make some negative
+        data_array_complex = 10*(1+1j-2*data_array_complex)
+        result_array_complex = array(
```

Comment:
Can you add a comment about where these veridical values come from (MATLAB, other package, etc.)

Reply:
@rgommers I can fix the other parts of this PR but I don't know how she set this up initially. This is true about all the values that are being used for verification.

Reply:
I suspect that @cynthia-rempel just took the output of the current `scipy.signal` functions here. Would be great if she could confirm, but let's not wait for that. If the tests pass I'm okay with adding them as pure regression tests and a comment to that effect (e.g. "testing against currently returned values as of scipy 1.1.0, for regression purposes").